src/app/_api/fetchGlobals.ts

```diff
@@ -1,102 +1,133 @@
-import type { Footer, Header, Settings } from '../../payload/payload-types'
-import { FOOTER_QUERY, HEADER_QUERY, SETTINGS_QUERY } from '../_graphql/globals'
-import { GRAPHQL_API_URL } from './shared'
+import type { Footer, Header, Settings, Social } from "../../payload/payload-types";
+import { FOOTER_QUERY, HEADER_QUERY, SETTINGS_QUERY, SOCIALS_QUERY } from "../_graphql/globals";
+import { GRAPHQL_API_URL } from "./shared";
 
 export async function fetchSettings(): Promise<Settings> {
-  if (!GRAPHQL_API_URL) throw new Error('NEXT_PUBLIC_SERVER_URL not found')
+  if (!GRAPHQL_API_URL) throw new Error("NEXT_PUBLIC_SERVER_URL not found");
 
-  const settings = await fetch(`${GRAPHQL_API_URL}/api/graphql`, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    cache: 'no-store',
-    body: JSON.stringify({
-      query: SETTINGS_QUERY,
-    }),
-  })
-    ?.then(res => {
-      if (!res.ok) throw new Error('Error fetching doc')
-      return res.json()
+  try {
+    return await fetch(`${GRAPHQL_API_URL}/api/graphql`, {
```

Comment:
Could we not reuse the fetchDoc function for these requests? Or at least have a util function to handle the fetch boilerplate.

Reply:
my plan down the road is to refactor all the fetchers using an util function, i've just been recycling for the time being to save time. if it makes most sense to refactor it now I can do it, what is the best approach/main priority?

Reply:
In that case, it's fine for now. Better to refactor when you know better what will be needed 👍